src/saga/mergeUtils.js

```diff
@@ -181,23 +28,48 @@ function simpleMerge(oRow, aRow, bRow, rowIndex, sheetName) {
 */
 export function simpleMerge2D(origin, aValues, bValues, sheetName) {
```

Comment:
Since our algorithms are changing quite dramatically quite quickly, it would be helpful for me if we start leaving more detailed comments that explain the logic of what is happening. Also, I think if we do a big refactor like this, in the PR, let's explain why we did it. It will be helpful for current and future understanding.